### PR TITLE
Assorted backports for 2.2.x

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -257,7 +257,7 @@ jobs:
           . ~/virtualenvs/pandas-dev/bin/activate
           python -m pip install --no-cache-dir -U pip wheel setuptools meson[ninja]==1.2.1 meson-python==0.13.1
           python -m pip install numpy --config-settings=setup-args="-Dallow-noblas=true"
-          python -m pip install --no-cache-dir versioneer[toml] cython python-dateutil pytz pytest>=7.3.2 pytest-xdist>=2.2.0 hypothesis>=6.46.1
+          python -m pip install --no-cache-dir versioneer[toml] cython python-dateutil pytz<2024.2 pytest>=7.3.2 pytest-xdist>=2.2.0 hypothesis>=6.46.1
```
Comment (lithomas1) [inline .github/workflows/unit-tests.yml:260]: pytz 2024.2 is causing a failure in ``test_array.py``
(relating to constructing an array from datetime objects with pytz timezones)

I've pinned in testing for now.

I'm wondering if I should pin it as a dep too?
(as far as I can tell, all pytz did was update their IANA tzdb to 2024.2, so maybe this fixes itself when everyone else does too?)

**Comment by WillAyd** [comment]
On phone so sorry for not linking better, but I think the DTI failures need to be partially pulled from #55901. 

If you look at the commits in the 2.3 branch I started you will see a commit where I do that

**Comment by lithomas1** [comment]
> On phone so sorry for not linking better, but I think the DTI failures need to be partially pulled from #55901.
> 
> If you look at the commits in the 2.3 branch I started you will see a commit where I do that

I think I have most of the relevant fixes picked from your PR, actually.
(and some of the rest I will fix by backporting some other PRs).

Only headscratcher for me right now is the ARM build jobs (which also happen to fail on main).
Interesting that it's only the build and not the test job, though ....

**Comment by lithomas1** [comment]
all green 🚀

**Pull request merged**